solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1239 @@
+			return false;
+		}
+
+		m_options.yulOptimiserSteps = m_args[g_strYulOptimizations].as<string>();
+	}
+
+	if (m_args.count(g_argImportAst) > 0)
+		m_options.inputMode = InputMode::CompilerWithASTImport;
+	else
+		m_options.errorRecovery = (m_args.count(g_argErrorRecovery) > 0);
+
+	solAssert(m_options.inputMode == InputMode::Compiler || m_options.inputMode == InputMode::CompilerWithASTImport, "");
+	return true;
+}
+
+bool CommandLineParser::hasOutput()
+{
+	return g_hasOutput;
+}
+
+bool CommandLineParser::parseCombinedJsonOption()
+{
+	if (!m_args.count(g_argCombinedJson))
+		return true;
+
+	set<string> requests;
+	for (string const& item: boost::split(requests, m_args[g_argCombinedJson].as<string>(), boost::is_any_of(",")))
+		if (!g_combinedJsonArgs.count(item))
+		{
+			serr() << "Invalid option to --" << g_argCombinedJson << ": " << item << endl;
+			return false;
+		}
+
+	m_options.combinedJsonRequests = CombinedJsonRequests{};
+	m_options.combinedJsonRequests->abi = (requests.count(g_strAbi) > 0);
+	m_options.combinedJsonRequests->metadata = (requests.count("metadata") > 0);
+	m_options.combinedJsonRequests->binary = (requests.count(g_strBinary) > 0);
+	m_options.combinedJsonRequests->binaryRuntime = (requests.count(g_strBinaryRuntime) > 0);
+	m_options.combinedJsonRequests->opcodes = (requests.count(g_strOpcodes) > 0);
+	m_options.combinedJsonRequests->asm_ = (requests.count(g_strAsm) > 0);
+	m_options.combinedJsonRequests->storageLayout = (requests.count(g_strStorageLayout) > 0);
+	m_options.combinedJsonRequests->generatedSources = (requests.count(g_strGeneratedSources) > 0);
+	m_options.combinedJsonRequests->generatedSourcesRuntime = (requests.count(g_strGeneratedSourcesRuntime) > 0);
+	m_options.combinedJsonRequests->srcMap = (requests.count(g_strSrcMap) > 0);
+	m_options.combinedJsonRequests->srcMapRuntime = (requests.count(g_strSrcMapRuntime) > 0);
+	m_options.combinedJsonRequests->funDebug = (requests.count(g_strFunDebug) > 0);
+	m_options.combinedJsonRequests->funDebugRuntime = (requests.count(g_strFunDebugRuntime) > 0);
+	m_options.combinedJsonRequests->signatureHashes = (requests.count(g_strSignatureHashes) > 0);
+	m_options.combinedJsonRequests->natspecDev = (requests.count(g_strNatspecDev) > 0);
+	m_options.combinedJsonRequests->natspecUser = (requests.count(g_strNatspecUser) > 0);
+	m_options.combinedJsonRequests->ast = (requests.count(g_strAst) > 0);
+
+	return true;
+}
+
+size_t CommandLineParser::countEnabledOptions(vector<string> const& _optionNames) const
+{
+	size_t count = 0;
+	for (string const& _option: _optionNames)
+		count += m_args.count(_option);
+
+	return count;
+}
+
+string CommandLineParser::joinOptionNames(vector<string> const& _optionNames, string _separator)
+{
+	return boost::algorithm::join(
+		_optionNames | ranges::views::transform([](string const& _option){ return "--" + _option; }),
+		_separator
+	);
```

Comment:
I replaced `boost::algorithm::join()` with our `joinHumanReadable()`.

By anonymous function you mean one in an anonymous namespace? In #11520 I move most of these namespaced functions into the class anyway because they need to access `serr()` which becomes a method too. This one does not use `serr()` but I'd rather leave it in the class too for consistency.

Overall, one thing that I don't like about putting stuff in these anonymous namespaces is that our cpp files are often huge and it's sometimes not obvious that this stuff is even there. Especially when someone puts it in the middle of the file rather than at the top. Putting it in the class has some downsides (changes in the header cause unnecessary recompilation) but it's also more discoverable that way. Also, not having it in the header makes it impossible to test directly.